src/merkle/tiered_smt/mod.rs

```diff
@@ -275,6 +323,7 @@ impl TieredSmt {
         }
 
         // update the root
+        self.nodes.insert(NodeIndex::root(), node);
```

Comment:
This is definitely more consistent, but OTOH we should always fetch the root node from `self.root` instead of querying the map

Reply:
The reason for this is actually to make `inner_node()` iteration easier. Without this, we'd have one less inner node than we'd like.